packages/aws-cdk-lib/aws-cloudwatch/lib/variable.ts

```diff
@@ -0,0 +1,195 @@
+export enum VariableInputType {
+  /**
+   * Freeform text input box
+   */
+  INPUT = 'input',
+
+  /**
+   * A dropdown of pre-defined values, or values filled in from a metric search query
+   */
+  RADIO = 'radio',
+
+  /**
+   * A set of pre-defined radio buttons, which can also be defined from a metric search query
+   */
+  SELECT = 'select',
+}
+
+export enum VariableType {
+  /**
+   * A property variable changes the values of all instances of a property in the list of widgets in the dashboard.
+   */
+  PROPERTY = 'property',
+
+  /**
+   * A pattern variable is one that changes a regex pattern across the dashboard JSON
+   */
+  PATTERN = 'pattern',
+}
+
+/**
+ * A single dashboard variable
+ */
+export interface IVariable {
```

Comment:
I didn't notice before, but why are we naming this `IVariable`? the `IXxx` naming convention has meaning in cdk land. It usually serves as a way for users to specify internal or external resources (i.e. `fromVariableArn` or something). That doesn't seem to be the case here.

>Therefore, when constructs are referenced anywhere in the API (e.g. in properties or methods of other resources or higher-level constructs), the resource interface (IFoo) should be used over concrete resource classes (Foo). This will allow users to supply either internal or external resources [awslint:ref-via-interface].

[Design Guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#construct-interface)

Reply:
Yeah. I tried to rename it in an earlier commit to `Variable` but I got jsii build errors.

I even tried to make it `DashboardVariable` and rename `DashboardVariable` to `DashboardVariableBase` but I got another error that since the interface has a behavior, it needs to follow the `Ixx` convention.

Any thoughts/guidance here?

Reply:
Well, I don't think its the end of the world for it to be `IVariable`. If jsii says so, lets just keep it that way.